src/vs/workbench/contrib/outline/browser/outlinePanel.ts

```diff
@@ -294,6 +300,15 @@ export class OutlinePanel extends ViewletPanel {
 		}
 	}
 
+	private outlineCollapseAllCommand: ICommandHandler = accessor => {
+		const actions = this.getActions();
+		for (const action of actions) {
+			if (action.id === 'collapse') {
```

Comment:
why?

Reply:
I'm not sure I understand which part or parts it is referring to.

If there is an easier way to call the collapse action, I am not familiar and would be more than happy to make changes.

If it is referring to the use of the ICommandHandler, this was the way I had found for registering a keybind.

Again, I am happy to adjust for feedback, however I am not sure this question is actionable on it's own. Regardless, thank you for your time.

Reply:
Having a command just to find another command/action to invoke it is always somewhat fishy. 

Take a look at this: https://github.com/Microsoft/vscode/blob/b513141f5d7b0728b3dfa2e0637c42a71351714d/src/vs/workbench/browser/actions/listCommands.ts#L255. That's the implementation for simple collapse and in that spirit collapse all should be done. Bonus: it will then work for all trees

Reply:
Thank you, Johannes. I definitely appreciate the pointer in the right direction.

I will commit a new set of changes very shortly. Meanwhile, I'd be happy to hear suggestions on the shortcut itself if the current one is not ideal. Any other feedback is certainly welcomed as well.